File: lib/Models/product.dart
// lib/models/product.dart

class Product {
  final int id;
  final String name;
  final int value;
  final DateTime createdAt;

  Product({
    required this.id,
    required this.name,
    required this.value,
    required this.createdAt,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      value: json['value'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}

File: MyApi/Models/Product.cs
using System;
using System.ComponentModel.DataAnnotations;
namespace MyApi.Models
{
    public class Product
    {
        public int Id { get; set; }
        [Required(ErrorMessage = "Id is required")]

        public string? Name { get; set; } 
        [Required(ErrorMessage = "Name is required")]

        public int Value { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime ModifiedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}
